packages/migration-graph-ember/src/utils/discover-ember-service-dependencies.ts

```diff
@@ -0,0 +1,224 @@
+import type {
+  CallExpression,
+  ClassDeclaration,
+  ClassExpression,
+  ClassMember,
+  ClassProperty,
+  Decorator,
+  ExportDefaultDeclaration,
+  Identifier,
+  ImportDeclaration,
+  ImportSpecifier,
+  ModuleItem,
+  NamedImportSpecifier,
+  StringLiteral,
+} from '@swc/core';
+import { parseFileSync } from '@swc/core';
+import debug from 'debug';
+import { join } from 'path';
+
+type EmberInferredServiceDependency = {
+  addonName?: string;
+  serviceName: string;
+};
+
+const DEBUG_CALLBACK = debug('rehearsal:discoverEmberServiceDependencies');
+
+const EMPTY_RESULT: EmberInferredServiceDependency[] = [];
+
+export function discoverServiceDependencies(
+  baseDir: string,
+  pathToFile: string
+): EmberInferredServiceDependency[] {
+  const filePath = join(baseDir, pathToFile);
+
+  const parsed = parseFileSync(filePath, { syntax: 'ecmascript', decorators: true });
+
+  const maybeImportDeclaration = findImportDeclarationsBySource(parsed.body, '@ember/service');
+
+  if (!maybeImportDeclaration) {
+    DEBUG_CALLBACK('No import declaration found');
+    return EMPTY_RESULT;
+  }
+
+  const importDeclaration = maybeImportDeclaration as ImportDeclaration;
+
+  // In the case of re-assignment from ineject as service, we walk the specifiers
+  const maybeSpecifier = findSpecifierByExportedName(importDeclaration.specifiers, 'inject');
+
+  if (!maybeSpecifier) {
+    return EMPTY_RESULT;
+  }
+
+  const decoratorName = maybeSpecifier.local.value;
+
+  if (!decoratorName) {
+    return EMPTY_RESULT;
+  }
+
+  let foundClasses: Array<ClassExpression | ClassDeclaration> = [];
+
+  // Find all ClassDeclaration in body of the parsed file
+  const classDeclrations = findClassDeclarations(parsed.body);
+
+  if (classDeclrations) {
+    foundClasses = foundClasses.concat(Array.from(classDeclrations));
+  }
+
+  // Typical use case will have ExportDefaultDeclaration where the body is a ClassExpression
+  const someDefaultExport = parsed.body.find(
+    (m: ModuleItem) => m.type == 'ExportDefaultDeclaration'
+  );
+
+  if (someDefaultExport) {
+    const edd = someDefaultExport as ExportDefaultDeclaration;
+    // Check if the ExportDefaultDeclaration has a ClassExpression
+    if (edd.decl.type === 'ClassExpression') {
+      foundClasses.push(edd.decl as ClassExpression);
+    } else {
+      DEBUG_CALLBACK('Default export does not contain a ClassExpression');
+    }
+  }
+
+  if (foundClasses.length < 1) {
+    throw new Error('Unexpected Error: No classes found despite having imported @ember/service.');
```

Comment:
I don't think we want to throw in the case. Probably should just return out with an empty result.